task_manager/app/models/task.rb

```diff
@@ -21,10 +23,11 @@ class Task < ApplicationRecord
   validates :priority, inclusion: { in: Task.priorities.keys }
   validates :progress, inclusion: { in: Task.progresses.keys }
 
-  scope :search_name, -> (keyword_name) { where(['name like?', "%#{keyword_name}%"]) }
+  scope :search_name, -> (keyword_name) { where(['tasks.name like?', "%#{keyword_name}%"]) }
   scope :search_progress, -> (keyword_progress) { where(progress: Task.progresses[keyword_progress]) if keyword_progress.present? }
   scope :search_user_id, -> (user_id) { where(user_id: user_id) }
   scope :sort_column_direction, -> (column, direction) { order(sort_column(column) => sort_direction(direction)) }
+  scope :search_label_id, -> (keyword_label_id) { where(labels: { id: keyword_label_id }) if keyword_label_id.present? }
```

Comment:
このスコープ、単体で利用するとエラーになる = join必須なので、
joinも組み込んだほうがよさそうです。
※controller側からincludesが必要となったとして、joins includes が重複しても問題ないです

```
irb(main):003:0> Task.search_label_id(1)
  Task Load (0.7ms)  SELECT `tasks`.* FROM `tasks` WHERE `labels`.`id` = 1 /* loading for inspect */ LIMIT 11
Traceback (most recent call last):
ActiveRecord::StatementInvalid (Mysql2::Error: Unknown column 'labels.id' in 'where clause')
```


```suggestion
  scope :search_label_id, -> (keyword_label_id) { joins(:labels).where(labels: { id: keyword_label_id }) if keyword_label_id.present? }
```

Reply:
今のscopeのままでも全然大丈夫なのですが、
参考までにcontroller側で↓みたいなこともできます。

```ruby
Task.joins(:labels).merge(Label.where(id: 1))
```

Reply:
ありがとうございます！
こちらの提案を承認させてもらいました！僕も、このメソッド単体で考えた際に、動くものではないなと感じていたため、
scopeにjoinsを入れ込むことにしました！
joinとincludeが重なっても、いい感じにsql文を作ってくれるんですね、、、！

以下のやつにすると、labelのscopeでかけるので、責務的には下のほうがいいとも思いました！
ありがとうございます！
`Task.joins(:labels).merge(Label.where(id: 1))`